nomenclature/processor/generic.py

```diff
@@ -0,0 +1,153 @@
+    def validate_components(cls, v, info: ValidationInfo):
+        # components have to be unique for creating rename-mapping (component -> target)
+        all_components = list()
+        for item in v:
+            all_components.extend(item.components)
+        _validate_items(all_components, info, "Duplicate component")
+        return v
+
+    @field_validator("mapping")
+    def validate_target_vs_components(cls, v, info: ValidationInfo):
+        # guard against having identical target and component
+        _codes = list()
+        for item in v:
+            _codes.append(item.name)
+            _codes.extend(item.components)
+        _validate_items(_codes, info, "Non-unique target and component")
+        return v
+
+    @property
+    def codes(self):
+        _codes = list()
+        for item in self.mapping:
+            _codes.append(item.name)
+            _codes.extend(item.components)
+        return _codes
+
+    def validate_with_definition(self, dsd: DataStructureDefinition) -> None:
+        error = None
+        # check for codes that are not defined in the codelists
+        codelist = getattr(dsd, self.dimension, None)
+        # no validation if codelist is not defined or filter-item is None
+        if codelist is None:
+            error = f"Dimension '{self.dimension}' not found in DataStructureDefinition"
+        elif invalid := codelist.validate_items(self.codes):
+            error = (
+                f"The following {self.dimension}s are not defined in the "
+                "DataStructureDefinition:\n - " + "\n - ".join(invalid)
+            )
+        if error:
+            raise ValueError(error + "\nin " + str(self.file) + "")
+
+    @classmethod
+    def from_file(cls, file: Path | str):
+        """Initialize an AggregatorMapping from a file.
+
+        .. code:: yaml
+
+        dimension: <some_dimension>
+        aggregate:
+          - Target Value:
+            - Source Value A
+            - Source Value B
+
+        """
+        file = Path(file) if isinstance(file, str) else file
+        try:
+            with open(file, "r", encoding="utf-8") as f:
+                mapping_input = yaml.safe_load(f)
+
+            mapping_list: list[dict[str, list]] = []
+            for item in mapping_input["aggregate"]:
+                # TODO explicit check that only one key-value pair exists per item
+                mapping_list.append(
+                    dict(name=list(item)[0], components=list(item.values())[0])
+                )
+        except Exception as error:
+            raise ValueError(f"{error} in {get_relative_path(file)}") from error
+        return cls(
+            dimension=mapping_input["dimension"],
+            mapping=mapping_list,  # type: ignore
+            file=get_relative_path(file),
+        )
+
+
+def _validate_items(items, info, _type):
+    duplicates = [item for item, count in Counter(items).items() if count > 1]
+    if duplicates:
+        raise PydanticCustomError(
```

Comment:
Even though I introduced them, I'd actually say we should move away from using custom errors. The way that I see it, they don't really provide any additional benefit and just make the code harder to read. 
We don't catch specific errors at any higher level and even if we did, they get wrapped in some pydantic error anyway

Reply:
Well, one benefit of the custom errors is that the code in the actual files is shorter, and the concatenation of a readable string with filename, etc. is moved elsewhere.

I agree that the error messages should be standardized and reused, but this PR is not the right place for that discussion.